graphql/util.go

```diff
@@ -435,21 +435,36 @@ func FilterTasksByBaseStatuses(taskResults []*TaskResult, baseStatuses []string,
 }
 func ConvertDBTasksToGqlTasks(tasks []task.Task, baseTaskStatuses BaseTaskStatuses) []*TaskResult {
 	var taskResults []*TaskResult
-	for _, task := range tasks {
-		t := TaskResult{
-			ID:           task.Id,
-			DisplayName:  task.DisplayName,
-			Version:      task.Version,
-			Status:       task.GetDisplayStatus(),
-			BuildVariant: task.BuildVariant,
-			Blocked:      task.Blocked(),
-			Aborted:      task.Aborted,
+	for _, t := range tasks {
+		result := TaskResult{
+			ID:           t.Id,
+			DisplayName:  t.DisplayName,
+			Version:      t.Version,
+			Status:       t.GetDisplayStatus(),
+			BuildVariant: t.BuildVariant,
+			Blocked:      t.Blocked(),
+			Aborted:      t.Aborted,
 		}
-		if baseTaskStatuses != nil && baseTaskStatuses[task.BuildVariant] != nil {
-			baseStatus := baseTaskStatuses[task.BuildVariant][task.DisplayName]
-			t.BaseStatus = &baseStatus
+		if baseTaskStatuses != nil && baseTaskStatuses[t.BuildVariant] != nil {
+			baseStatus := baseTaskStatuses[t.BuildVariant][t.DisplayName]
+			result.BaseStatus = &baseStatus
 		}
-		taskResults = append(taskResults, &t)
+		if len(t.ExecutionTasksFull) > 0 {
```

Comment:
we can get ExecutionTasksFull by building `t` from service. I hope this allows us to remove the for loop on line 454